### DISCUSSION TITLE
explaining 'addByPrefix' and 'addByIndices'

### DISCUSSION THREAD

**Discussion by BernardoGP4504** [YoshiCrafter29/YoshiCrafterEngine#225] [Show and tell]
Let's just say that you want to make a character using Haxe or just wanted to know what are those (if you ever looked at the files).
_Then this is the right source to know!_

### addByPrefix('animation:String', 'xml anim:String', framerate:Int, loop:Bool);

an **`addbyprefix`** is adding a normal animation for your character for example, let's add gf's left animation:
```
character.animation.addByPrefix('singLEFT', 'GF left note', 24, false);
```

animation: the animation
xml anim: that's the animation in the `.xml`
framerate: self explanatory
loop: if your not going to loop the animation then use **`false`**. If you want it to loop then don't use this field like:
```
character.animation.addByPrefix('scared', 'GF FEAR', 24);
```

### addByIndices('animation:String', 'xml anim:String', [indices:String], "", framerate:Int, loop:Bool);
basically the same as **addByPrefix** but is uses indices but, what are indices?

## Indices
They are basically each frame of the animation, for an better explaination here's an example:
This is Edd, he is a Eddsworld character and a character from the FNF Funkin' Eddventure mod. Let's take a look in his left animation
![left0000](https://user-images.githubusercontent.com/101405920/178606840-7fa40ff9-a0c8-4490-81b0-c0819025f86b.png)
**This is the first _indice_ from his idle (0)**

![left0002](https://user-images.githubusercontent.com/101405920/178606848-b96fea55-8974-42f4-a663-738b0fcddaaa.png)
**This is the third _indice_ from his idle (2)**
did you spot the difference? They are separate frames; ***frames are nothing more than indices***
